app/src/app.rs

```diff
@@ -81,13 +214,20 @@ pub async fn initialize_state_from_existing_block(
             //     requisite data for the validation is missing
             debug!("Payload is valid");
             debug!("latest block {:?}", state.latest_block);
+
+            // Read the validator set at the stored block - this is the validator set
+            // that will be active for the NEXT height (where consensus will start)
             let block_validator_set = read_validators_from_contract(
                 engine.eth.url().as_ref(),
                 &latest_block_candidate_from_store.block_hash,
             )
             .await?;
-            debug!("🌈 Got block validator set: {:?}", block_validator_set);
-            state.set_validator_set(start_height, block_validator_set);
+
+            // Consensus will start at the next height, so we set the validator set for that height
```

Comment:
@jmalicevic I also fixed the code here. The validator set from the latest accepted block was stored for the wrong height (current instead of next), causing the code to fail later since the validator set for the next height is requested.